three-tier-main.tf

```diff
@@ -36,7 +36,7 @@ resource "aws_subnet" "web-subnet-2" {
   vpc_id                  = aws_vpc.my-vpc.id
   cidr_block              = "10.0.2.0/24"
   availability_zone       = "ap-southeast-2b"
-  map_public_ip_on_launch = true
+  map_public_ip_on_launch = false
```

Comment:
**Tenable.cs remediation**
 
**[MEDIUM]: Ensure public IP on launch is not enabled for AWS Subnets**
 
**Vulnerability**
Network interfaces launched in the subnet will automatically receive public IP address. Thus, exposing the instances attached to the interface to the public, provided other conditions to make instance public satisfy.
 
**Remediation**
All subnets have an attribute that determines whether a network interface created in the subnet automatically receives a public IPv4 address. Therefore, it is recommended to set it to FALSE. In AWS Console - 
 1. Sign in to the AWS Console and go to the VPC dashboard.
 2. In the navigation pane, select subnets.
 3. Select your subnet and then choose Subnet Actions, Modify auto-assign IP settings.
 4. Clear the Enable auto-assign public IPv4 address check box and then choose Save.
 In terraform - Modify 'aws_subnet' resource and set 'map_public_ip_on_launch' to false.

source: tenablecs-